CMakeLists.txt

```diff
@@ -5,15 +5,37 @@ set(PACKAGE_PROJECT_ROOT_PATH
     CACHE INTERNAL "The path to the PackageProject directory"
 )
 
+# Add custom TO_INSTALL property to the project targets
+define_property(
+    TARGET
+    PROPERTY TO_INSTALL
+    BRIEF_DOCS ""
+    FULL_DOCS ""
+)
+
+# Function to get all project targets
+function (get_all_cmake_targets out_var current_dir)
+    get_property(targets DIRECTORY ${current_dir} PROPERTY BUILDSYSTEM_TARGETS)
+    get_property(subdirs DIRECTORY ${current_dir} PROPERTY SUBDIRECTORIES)
+
+    foreach(subdir ${subdirs})
+        get_all_cmake_targets(subdir_targets ${subdir})
+        list(APPEND targets ${subdir_targets})
+    endforeach()
```

Comment:
Does it make sense to install targets from all subdirectories as well? This could accidentally result in unrelated dependencies being installed as part of the main package.

Reply:
Only the targets marked with TO_INSTALL property will be installed, so this should be safe.

Reply:
If some nested directory uses dependency, which sets this property forcefully than yes, it will be installed, but then this is not a well-behaving dependency.